service/metadata/service/object.go

```diff
@@ -2,59 +2,107 @@ package service
 
 import (
 	"context"
-
-	model "github.com/bnb-chain/greenfield-storage-provider/store/bsdb"
-	"github.com/bnb-chain/greenfield/types/s3util"
+	"encoding/base64"
 
 	"cosmossdk.io/math"
+	"github.com/bnb-chain/greenfield/types/s3util"
 	"github.com/bnb-chain/greenfield/x/storage/types"
 
 	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
 	metatypes "github.com/bnb-chain/greenfield-storage-provider/service/metadata/types"
+	model "github.com/bnb-chain/greenfield-storage-provider/store/bsdb"
 )
 
 // ListObjectsByBucketName list objects info by a bucket name
 func (metadata *Metadata) ListObjectsByBucketName(ctx context.Context, req *metatypes.ListObjectsByBucketNameRequest) (resp *metatypes.ListObjectsByBucketNameResponse, err error) {
-	ctx = log.Context(ctx, req)
+	var (
+		results               []*model.ListObjectsResult
+		keyCount              uint64
+		isTruncated           bool
+		nextContinuationToken string
+		maxKeys               uint64
+		commonPrefixes        []string
+		res                   []*metatypes.Object
+	)
+
+	maxKeys = req.MaxKeys
+	// if the user does not provide any input parameters, default values will be used
+	if req.MaxKeys == 0 {
+		maxKeys = model.ListObjectsDefaultMaxKeys
+	}
+
+	// returns some or all (up to 1000) of the objects in a bucket with each request
+	if req.MaxKeys > model.ListObjectsLimitSize {
+		maxKeys = model.ListObjectsLimitSize
+	}
 
-	objects, err := metadata.bsDB.ListObjectsByBucketName(req.BucketName)
+	ctx = log.Context(ctx, req)
+	results, err = metadata.bsDB.ListObjectsByBucketName(req.BucketName, req.ContinuationToken, req.Prefix, req.Delimiter, int(maxKeys))
```

Comment:
req. StartAfter is unused?

Reply:
The "start_after" is used to specify the starting point of the query, while "next_continuation_token" is used to persist the query state for use in the next query. These two are not used simultaneously.

Within the code logic, if the next_continuation_token is empty, the "start_after" will assign the value to the marker (ContinuationToken). This approach is based on the logic found in the ListObjectV2 function of Minio (similar to AWS S3).

Reply:
If this field is not used, maybe it can be removed in the proto definition.

Reply:
"start_after" is being used and it is assigned a value to the data at the handler level, the "start_after" will assign the value to the marker (ContinuationToken).

```
if requestContinuationToken == "" {
            continuationToken = requestStartAfter
}
```
https://github.com/minio/minio/blob/b92cdea5789aab0e71f98c7dbdca9d5c8d22a6fe/cmd/erasure-server-pool.go#L1191